src/py/mat3ra/made/tools/analyze/rotation_analyzer.py

```diff
@@ -0,0 +1,174 @@
+        """
+        original_analyzer = BasisMaterialAnalyzer(material=original_material)
+        original_fingerprint = original_analyzer.get_material_fingerprint(layer_thickness)
+
+        current_analyzer = BasisMaterialAnalyzer(material=self.material)
+        current_fingerprint = current_analyzer.get_material_fingerprint(layer_thickness)
+
+        return self._detect_rotation_between_fingerprints(original_fingerprint, current_fingerprint, threshold)
+
+    def _detect_rotation_between_fingerprints(
+        self,
+        original_fingerprint: MaterialFingerprintAllAxes,
+        current_fingerprint: MaterialFingerprintAllAxes,
+        threshold: float = 0.1,
+    ) -> RotationDetectionResult:
+        """
+        Detect rotation between two material fingerprints.
+
+        Args:
+            original_fingerprint: Original material fingerprint
+            current_fingerprint: Current material fingerprint
+            threshold: Minimum improvement threshold to consider a rotation detected
+
+        Returns:
+            RotationDetectionResult: Rotation detection results
+        """
+        direct_score = self._calculate_alignment_score(
+            original_fingerprint, current_fingerprint, rotation_matrix=np.eye(3)
+        )
+
+        rotation_candidates = self._generate_rotation_candidates()
+
+        best_score = direct_score
+        best_result = RotationDetectionResult(
+            is_rotated=False,
+            rotation_matrix=None,
+            rotation_angle=None,
+            rotation_axis=None,
+            confidence=direct_score,
+        )
+
+        for rotation_matrix, angle, axis in rotation_candidates:
+            score = self._calculate_alignment_score(original_fingerprint, current_fingerprint, rotation_matrix)
+
+            if score > best_score + threshold:
+                best_score = score
+                best_result = RotationDetectionResult(
+                    is_rotated=True,
+                    rotation_matrix=rotation_matrix,
+                    rotation_angle=angle,
+                    rotation_axis=axis,
+                    confidence=score,
+                )
+
+        return best_result
+
+    def _calculate_alignment_score(
+        self,
+        original_fingerprint: MaterialFingerprintAllAxes,
+        current_fingerprint: MaterialFingerprintAllAxes,
+        rotation_matrix: np.ndarray,
+    ) -> float:
+        """
+        Calculate alignment score between fingerprints with rotation.
+
+        Args:
+            original_fingerprint: Original material fingerprint
+            current_fingerprint: Current material fingerprint
+            rotation_matrix: Rotation matrix to apply before comparison (use np.eye(3) for no rotation)
+
+        Returns:
+            float: Alignment score (0.0 to 1.0)
+        """
+        total_score = 0.0
+        count = 0
+
+        for i, self_axis in enumerate(MaterialFingerprintAllAxes.ALL_AXES):
```

Comment:
This can be replaced with matrix multiplication. If reverse needs to be checked, we should use mirror transformation.